### DISCUSSION TITLE
Missing branch in truth record of Delphes root file

### DISCUSSION THREAD

**Discussion by jonas-botz** [CheckMATE2/checkmate2#17] [Q&A]
Hey, I want to use the root output file of Delphes as used from CheckMATE to analyse the truth record coming from Pythia. I am expecting that Pythia just passes this to Delphes and it's saved inside the root output file as a branch in the Delphes tree. However in the "particle" branch, there are only quarks, taus and gluons. It seems like there are many truth particles missing. Maybe you can help!?

Thanks and Warm Regards

Jonas

**Comment by NishitaDesai** [comment]
Yes, CheckMATE by default keeps only those truth particles that are of use in further tagging.  The Long-lived particles beta branch should have the full truth-level data.  You may try using that version.

If you do not need CheckMATE’s functionality, why not try to get Pythia to write a HepMC file and pass that to Delphes directly to get your root file?

Best,
Nishita.
